src/nemos/basis/basis.py

```diff
@@ -1830,3 +1825,248 @@ def _check_window_size(self, window_size: int):
+            self,
+            n_basis_funcs=1,
+            label=label,
+        )
+
+    @add_docstring("evaluate_on_grid", OrthExponentialBasis)
+    def evaluate_on_grid(self, n_samples: int) -> Tuple[NDArray, NDArray]:
+        """
+        Examples
+        --------
+        >>> import numpy as np
+        >>> import matplotlib.pyplot as plt
+        >>> from nemos.basis import IdentityEval
+        >>> basis = IdentityEval()
+        >>> sample_points, basis_values = basis.evaluate_on_grid(100)
+
+        """
+        return super().evaluate_on_grid(n_samples=n_samples)
+
+    @add_docstring("_compute_features", EvalBasisMixin)
+    def compute_features(self, xi: ArrayLike) -> FeatureMatrix:
+        """
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from nemos.basis import IdentityEval
+
+        >>> # Generate data
+        >>> num_samples = 1000
+        >>> X = np.random.normal(size=(num_samples, ))  # raw time series
+        >>> basis = IdentityEval()
+        >>> features = basis.compute_features(X)  # basis transformed time series
+        >>> features.shape
+        (1000, 1)
+
+        """
+        return super().compute_features(xi)
+
+    @add_docstring("split_by_feature", IdentityBasis)
+    def split_by_feature(
+        self,
+        x: NDArray,
+        axis: int = 1,
+    ):
+        """
+        Examples
+        --------
+        >>> import numpy as np
+        >>> from nemos.basis import IdentityEval
+        >>> from nemos.glm import GLM
+        >>> # Define an additive basis
+        >>> basis = IdentityEval(label="feature")
+        >>> # Generate a sample input array and compute features
+        >>> inp = np.random.randn(20)
+        >>> X = basis.compute_features(inp)
+        >>> # Split the feature matrix along axis 1
+        >>> split_features = basis.split_by_feature(X, axis=1)
+        >>> for feature, arr in split_features.items():
+        ...     print(f"{feature}: shape {arr.shape}")
+        feature: shape (20, 1, 1)
+
+        """
+        return super().split_by_feature(x, axis=axis)
+
+    @add_docstring("set_input_shape", AtomicBasisMixin)
+    def set_input_shape(self, xi: int | tuple[int, ...] | NDArray):
+        """
+        Examples
+        --------
+        >>> import nemos as nmo
+        >>> import numpy as np
+        >>> basis = nmo.basis.IdentityEval()
+        >>> # Configure with an integer input:
+        >>> _ = basis.set_input_shape(3)
+        >>> basis.n_output_features
+        3
+        >>> # Configure with a tuple:
+        >>> _ = basis.set_input_shape((4, 5))
+        >>> basis.n_output_features
+        20
+        >>> # Configure with an array:
+        >>> x = np.ones((10, 4, 5))
+        >>> _ = basis.set_input_shape(x)
+        >>> basis.n_output_features
+        20
+
+        """
+        return AtomicBasisMixin.set_input_shape(self, xi)
+
+
+class HistoryConv(ConvBasisMixin, HistoryBasis):
+    """Identity basis function.
```

Comment:
Need a better one-liner here, at least to distinguish with IdentityEval

Reply:
yep, copy-paste issue